test-apps/nuxt/pages/login.vue

```diff
@@ -1,10 +1,9 @@
 <script lang="ts" setup>
-const { data } = await useFetch("/api/user");
-if (!data.value) throw createError("Failed to fetch data");
-const user = data.value.user;
-if (user) {
-	await navigateTo("/");
-}
+onBeforeMount(async () => {
+	const { data, error } = await useFetch<{ user: any }>("/api/user");
```

Comment:
useFetch should not be called in beforeMount - instead this could be moved to setup function with `server: false` option